src/sentry/integrations/discord/message_builder/base/component/action_row.py

```diff
@@ -0,0 +1,27 @@
+from __future__ import annotations
+
+from collections.abc import Iterable
+
+from .base import DiscordMessageComponent
+
+
+class DiscordActionRowError(Exception):
+    def __init__(self) -> None:
+        super().__init__(
+            "A DiscordActionRow cannot be contained within another DiscordActionRow's components"
+        )
+
+
+class DiscordActionRow(DiscordMessageComponent):
+    def __init__(self, components: Iterable[DiscordMessageComponent]):
+        for component in components:
+            if isinstance(component, DiscordActionRow):
+                raise DiscordActionRowError()
```

Comment:
Not sure if this is the best way to deal with this. Lmk if there's a better way to enforce this constraint. The goal of this is to give a future message builder more information than the 0 feedback Discord provides if you send a bad message.